client/layout/logged-out.jsx

```diff
@@ -127,7 +125,9 @@ const LayoutLoggedOut = ( {
 					{ secondary }
 				</div>
 			</div>
-			{ config.isEnabled( 'gdpr-banner' ) && <GdprBanner showGdprBanner={ showGdprBanner } /> }
+			{ config.isEnabled( 'gdpr-banner' ) && (
+				<AsyncLoad require="calypso/blocks/query-a8c-analytics-banners" placeholder={ null } />
```

Comment:
The logged out component is rendered on the server, so attempting to async load there it will fail on the server. While it will be rendered on the client, the problem before has been that doing it on the client the GDPR banner has caused a major LCP before, and that Core Web Vitals metrics have suffered from that. 

See https://github.com/Automattic/wp-calypso/pull/59848 for context and cc @jsnajdr and @sgomes as they worked on that before.

Reply:
The way the GDPR banner is created now is completely different and, in its nature, shouldn't impact LCP (as the component itself injects external script to the client DOM). Previously, we wanted to mitigate the time-consuming process of determining whether the banner should be displayed (correct me if I'm wrong).

Can/Should we inject this external script on the server side somehow? 
Will injecting an external script impact our LCP? (No, in my understanding)

cc @sgomes, if you could help me understand these performance aspects, I can work on the ideal solution for the currently given context.

Reply:
In any case, I believe we intentionally didn't async load the GDPR banner in the logged-out layout component - in order for it to be rendered on the initial SSR render and avoid any additional large paints later. Is that something we can try out?

Reply:
Just wanted to confirm what @tyxla said; the issue is that on mobile the banner can easily become the largest contentful paint element. As such, we want it to be server-side rendered, so that it has the least possible impact on the LCP metric.

Reply:
I see the issue now. Thanks for confirming as well @sgomes. It's a difficult one in that case. Previously, we could SSR the GDPR Banner because it was a local React component. 

In this approach, the rendering of the banner happens in the external script. To my understanding, it's impossible to execute external scripts that will interact with DOM structure from the server side in SSR (even if `<script src=".../a8c-analytics.js" />` will be included on the server, this JS will only be executed in the client browser)? So the solution itself has a significant downside in that terms.

Please let me know if you have any ideas on how to address it. One alternative I see is to create a fallback GDPR Banner component to be used in SSR and use the most recent ([a8c-analytics.js](D85697-code)) version on the client-rendered version.

The best case would be if we could export the component directly from [a8c-analytics.js](D85697-code), but it's only in the future plans to create an npm lib from it.

Reply:
I wonder if you have considered moving the library to the Calypso monorepo? Judging by the fact that it looks like it's currently being created, it might be a good time to make this decision. There's already a solid infrastructure there to help with making it an isolated package, and it would still be way easier to iterate on it and also include it in the Calypso client. What do you think?

Reply:
That's a good remark. We've planned to do it later, but given the circumstances, maybe we can handle making it an npm package in the calypso repo with the current refactor. 

We would need to ensure that we maintain backward compatibility with wpcom (probably my importing the project as npm lib there) - but I think it's a good idea to consider. How do you feel about it @gmovr? I'm not sure what's the best way to approach it now.

Reply:
As a heads up, we're building the library with a reusable Cookie Banner (to begin with) that will be compatible with SSR. We'll implement it locally in wp-calypso rather than depending on the execution of an external script:

https://github.com/Automattic/wp-calypso/pull/70246